### ISSUE TITLE
jshint: Undeclared variable or function 'require'

### ISSUE THREAD

**Issue by zilvinasu** [flycheck/flycheck#579]
Hey, there is some unexpected behavior from *Flycheck*. This seems to be alright by using jshint [outside](http://jshint.com/) Emacs and Flycheck

```JavaScript
/*jshint node: true*/
var util = require('util'),
    gulp = require('gulp'),
    less = require('gulp-less'),
    path = require('path'),
    uuid = require('node-uuid'),
    unirest = require('unirest');
```

Though Emacs reports **require** as undeclared variable. Any ideas?

**Comment by lunaryorn** [comment]
@zilvinasu What happens if you run `jshint` from the terminal? 

`/*jshint node: true*/` doesn't have any effect, by the way.

**Comment by zilvinasu** [comment]
It doesn't report any error if I run jshint from terminal on the file with the content above

**Comment by lunaryorn** [comment]
@zilvinasu Please show me the *exact* command the *exact* output.

Also, please run type `C-c ! C-c` in the Emacs buffer, and copy and paste the output in the resulting buffer.

**Comment by zilvinasu** [comment]
@lunaryorn the output of `C-c ! C-c` is

```
-*- mode: compilation; default-directory: "~/Downloads/" -*-
Compilation started at Sat Jan 31 00:36:32

jshint --checkstyle-reporter /Users/hoppipolla/Downloads/test.js
<?xml version="1.0" encoding="utf-8"?>
<checkstyle version="4.3">
</checkstyle>

Compilation finished at Sat Jan 31 00:36:32
```

not sure what else  have you meant by exact command and exact output?

**Comment by lunaryorn** [comment]
Whatever error you see does not come from Flycheck then.  Closing as invalid.

Do you use JS2 Mode?

**Comment by zilvinasu** [comment]
Thanks, seems to be caused by `js2-mode` for real. Just tried disabling it.

**Comment by JRigotti** [comment]
If you came here from google (as I did), this is indeed as @zilvinasu said a js2-mode error and the solution is to `js2-include-node-externs` to `t`. I customized this variable, but I guess the following snippet in your configuration might do the trick:
```
(setq js2-include-node-externs t)
```
